V2.0.2

dtreffenstaedt opened the pull request:
Pull request for the v2.0.2 release as almost all relevant issues have been fixed and closed.

dtreffenstaedt committed:
Merge pull request #36 from MuonPi/v2.0.1

V2.0.1

hangeza committed:
- added TCP message for MQTT inhibit functionality
- added scan parameter BIAS in GUI scan tab
- added enhanced data export for completed scan including statistical errors
- started to replace setDac signals with set Threshold and setBias signals in GUI since they are more descriptive

marvin5300 committed:
Added set_property(TARGET muondetector PROPERTY POSITION_INDEPENDENT_CODE 1) to all targets so pie hardening lintian warning should go away hopefully.

Commit:
Modified the MqttHandler class to use libmosquitto instead of paho-mqttpp

Commit:
Added mosquitto_loop_stop call to cleanup method

marvin5300 committed:
Muondetector login should now automatically create path for mqtt login save file.
This should fix a bug where login cannot create file because folder created by daemon does not exist
when daemon was not started for the first time yet.

hangeza committed:
added preselection of max statistics for scans in GUI/scan tab

hangeza committed:
Merge branch 'dev' of https://github.com/MuonPi/muondetector-cleaned into dev

Commit:
Renamed custom targets from daemon and gui in order to be able to build both simultaneously

Commit:
Modified CMakeLists.txt to be able to build both the daemon and the gui package at the same time

Commit:
Revert "Modified CMakeLists.txt to be able to build both the daemon and the gui package at the same time"

This reverts commit 4fd70704c84d6094cea63d0b51c16563235228dd.

Commit:
Renamed PROJECT_XXX_DIR/FILS variables in CMakeLists.txt to subdir specific names

Commit:
Prepared library/CMakeLists.txt for transfer to include layout

Commit:
Migrated library CMakeLists.txt to cmake/library.cmake

Commit:
Migrated gui/CMakeLists.txt to cmake/gui.cmake

Commit:
Moved daemon/CMakeLists.txt towards cmake/daemon.cmake

Commit:
Minor adjustments between all three .cmake files

Commit:
Added dependencies on muondetector-shared target

Commit:
Made muondetector-shared library and object library instead of shared

Commit:
Update variable names in add_executable for muondetector-gui target

Commit:
Added appropriate variables for package descriptions

Commit:
More fixes to still wrong variable names

Commit:
Modified copyright and changelog file locations

Commit:
Slight fixes to file install locations

Commit:
Fixed parameter for mkdir command in custom target

Commit:
Added option to set the debian package filename

Commit:
Adjusted package naming scheme to be more consistent with debian package names

Commit:
Debian package name now uses the proper architecture name instead of the processor name

Commit:
Finally updated to proper debian package names

Commit:
Added package building option for building without daemon

Commit:
Added option for the proper package buildig for when only the daemon is built

Commit:
Cleaned up redundant CMake code a bit

Commit:
Changed set_property(TARGET ... to set_target_properties(...

Commit:
Fixed dependencies in CMakeLists.txt for daemon target

Commit:
Fixed data and log topics not being initialised correctly

Commit:
Fixed typo where an operator += should have been an operator =

dtreffenstaedt committed:
Merge pull request #37 from MuonPi/convert-libmosquitto

migrate from paho-mqttpp to libmosquitto

Commit:
Merge branch 'dev' into modify-cmake-for-packaging

Commit:
Removed last remnant of paho dependencies

dtreffenstaedt committed:
Merge pull request #43 from MuonPi/modify-cmake-for-packaging

Modify CMake structure to allow for simpler build process

Commit:
Added option to cmake project to build the daemon whcih gets set depending on the system processor

Commit:
Merge branch 'main' into dev

dtreffenstaedt committed:
Update README.md

Commit:
Split the mqtt library part from the main library

dtreffenstaedt committed:
Update daemon.cmake

Commit:
Removed unnecessary dependency in gui cmake file

Commit:
Made not building the daemon the default

Commit:
changed reconnect timer firing to work over signals and slots mechanism

andreSai committed:
Anpassung gui.cmake für MacOS
Hinzufügen APP-Icon für MacOS

andreSai committed:
Anpassung Darstellung Tab Parameter

Commit:
Changed ip address save file name and location for the gui

Commit:
Fixed forward declaration of class GnssSatellite to be correct

Commit:
Removed unused variables from gui

Commit:
Merge branch 'dev' into v2.0.2

Commit:
Changed old, deprecated QButtonGroup::buttonClicked over to new QButtonGroup::idClicked signals

Commit:
Simplified gui.cmake file if statement

Commit:
Merge branch 'andre' into v2.0.2

Commit:
Moved redundant code from .cmake files to common CMakeLists.txt

Commit:
Updated Version number to 2.0.2

Commit:
modified check for building the debian package to also do a negative check for APPLE instead of only for WIN32

Commit:
Added configuration options maxGeohashLength, store_local and tcp_port

andreSai committed:
fixed error in gui loadsettings

Commit:
Removed large amounts of commented out code

Commit:
Added missing space to the maxGeohashLength message

andreSai committed:
Update .gitignore

andreSai committed:
Create Info.plist.in

andreSai committed:
Update gui.cmake

andreSai committed:
Support for OSX Retina resolution

Commit:
Revert "Changed old, deprecated QButtonGroup::buttonClicked over to new QButtonGroup::idClicked signals"

This reverts commit fb130268c0b045b10d79879d61df795ab84fc00b.

dtreffenstaedt committed:
Merge pull request #44 from MuonPi/v202_andre_hdpi

Update .gitignore

hangeza committed:
Added restoration of origial values of selected scan parameters after a scan is finished.

Commit:
Added slot to QtSerialUblox that should close the serial port. This will be called when the daemon receives a SIGINT, SIGHUP or SIGTERM

Commit:
Merge branch 'v2.0.2' of github.com:MuonPi/muondetector into v2.0.2

Commit:
Added option to be able to edit the daemon in a non raspberrypi system

Commit:
Added additional version information for version string and added lines which send the version string to the server

hangeza committed:
Bugfix: implemented signal to change the gain setting when the checkbox in "Parameters" tab in the GUI is clicked

hangeza committed:
replaced the "connect" calls in constructor of ParamMonitorForm with the coresponding version taking the signals as argument instead of lambdas

dtreffenstaedt committed:
Merge pull request #46 from MuonPi/dev

Bugfix: implemented signal to change the gain setting when the checkb…

Commit:
Added flag to CMakeLists.txt which indicates whether the project is built on a raspberry pi

hangeza committed:
Deactivated pulse height plot in overview tab by default on startup.
Few clean-ups of commented-out lines.

closes #49

hangeza committed:
- Added display of current scan variable and measurement value including error in GUI's scan tab.
- Deleted some commented-out lines.

hangeza committed:
small bugfix: parameter scan progress bar range and stepping corrected

hangeza committed:
fixed correct enabling/disabling of plot widgets in GUI

Commit:
Improved output for muondetector-login program to make it more clear which input is expected.

Commit:
Removed unnecessary braces.

Commit:
Modified daemon class to use configuration object

Commit:
Removed unnecessary comparison of boolean with true

Commit:
Changed old-style iterator for loops to range-based for loops

Commit:
Merge branch 'dev' into v2.0.2

Commit:
Added connecting and connection error indicator to GUI

Commit:
Removed -dev addition from version number

Commit:
Properly disabled entire GUI in disconnected state.

Commit:
Improved connection indicator in gui

Commit:
Fixed systemd service file

Commit:
Removed unnecessary line from main window

Commit:
Fixed nonexistant and empty signals for calibform

Commit:
Cleaned up QML file

Commit:
Updated GUI to use QwtSlider and QDoubleSpinBox instead of QLineEdit and QSlider

hangeza commented on source/cmake/daemon.cmake:
```diff
@@ -0,0 +1,193 @@
+target_include_directories(muondetector-login PUBLIC
+    $<BUILD_INTERFACE:${LIBRARY_INCLUDE_DIR}>)
+
+target_link_libraries(muondetector-login
```
I have to include pthread to the target_link_libraries directive. Why it is missing now (even since some time in the past)?

dtreffenstaedt replied:
Fixed with https://github.com/MuonPi/muondetector/commit/2f288cc0a3a4f6b65f65f69a895f3d201b8292ad#diff-41cf6794ba4200b839c53531555f0f3998df4cbb01a4d5cb0b94e3ca5e23947d

hangeza commented on source/daemon/include/daemon.h:
```diff
@@ -129,11 +127,35 @@ class Daemon : public QTcpServer
-        bool new_configGnss, unsigned int new_eventTrigger, QString new_PeerAddress, quint16 new_PpeerPort,
-        QString new_serverAddress, quint16 new_serverPort, bool new_showout, bool new_showin, bool preamp1, bool preamp2, bool gain, QString station_ID,
-        bool new_polarity1, bool new_polarity2, QObject *parent = 0);
+    struct configuration {
```
Very good idea to replace the endless list of constructor arguments with a configuration struct. Which config parameters would we need additionally and could one use the same struct for config file interaction? Nothing to be solved with this release, just to keep it in mind to take care of for the next one.

hangeza commented on source/daemon/src/daemon.cpp:
```diff
@@ -6,10 +6,9 @@
-//#include <calib_struct.h>
 #include <ublox_messages.h>
 #include <tcpmessage_keys.h>
-#include <iomanip>      // std::setfill, std::setw
```
Why delete this comment? I find it quite useful to see for which functions a header is included...

hangeza commented on source/daemon/src/daemon.cpp:
```diff
@@ -2494,28 +2247,6 @@ void Daemon::onLogParameterPolled(){
         qDebug() << " file size: " << fileHandler->dataFileInfo().size()/(1024*1024) << "MiB";
     }
 
-//      Since Linux 2.3.23 (i386) and Linux 2.3.48 (all architectures) the
```
This comment block was intentionally copied from the kernel docs here in order to see the structure and the other available fields. Imho it serves some useful purpose, so i would ask to leave it.

dtreffenstaedt replied:
Fixed with https://github.com/MuonPi/muondetector/commit/246663e4ff2e27037d5c34e004d0f7785ca42e01#diff-41cf6794ba4200b839c53531555f0f3998df4cbb01a4d5cb0b94e3ca5e23947d

hangeza commented on source/gui/include/scanform.h:
```diff
@@ -9,39 +9,48 @@
-const static QVector<QString> SP_NAMES = { "VOID", "THR1", "THR2" };
+const static QVector<QString> SP_NAMES = { "VOID", "THR1", "THR2", "BIAS", "DAC4" };
 // list of possible observables
-const static QVector<QString> OP_NAMES = { "VOID", "UBXRATE" /*, "GPIORATE"*/ };
```
I would leave the commented-out GPIORATE here, since this would be the next scan observable to be implemented and is therefore helpful as a reminder

dtreffenstaedt replied:
Instead, use issues on github.

Stumbling on this comment will only happen by coincidence, and it is not clear why this is commented out or what it is supposed to mean.

If comments are used as a reminder, use a comment in the form of
`//TODO: Implement GPIORATE scan observable`
Those can be scanned and displayed by an IDE or other tools and offer useful information.

hangeza reviewed (approved):
General approval for the merge. I left some comments, though, mostly mocking the deletion of some imho useful comments inside the code.

Commit:
Applied clang-format

Commit:
Added pthread to linked libraries for daemon

Commit:
Added clang-format target to CMakeLists.txt

Commit:
Manually fixed formatting of glcdfont.c file

Additionally moved and renamed it more appropriately as a header file

Commit:
Added hack so the code compiles after clang-format

This is due to the include order of a few files in the project.
This is by no means a permanent fix and needs to be addressed.

hangeza committed:
Some clean-up and simplification of distributed code:
- Moved the implementation part of Histogram class to library/histogram.cpp. Before everything was in histogram.h only
- Added rescale functions to Histogram class. This obsoletes two functions which did the job up to now in Daemon class
- Included the system's i2c-dev headers (linux/i2c-dev.h) instead of a local copy. Included this only in the files where the header is needed, i.e. x9119.cpp and i2cdevice.cpp

Commit:
Removed obsolete source and header files

Commit:
Modified directory structure to make it clearer

i2c devices and spi devices now share a common root directory with their
own respective subdirectories

fixed fileencoding for sht21 class

Commit:
Applied clang-format

Commit:
Fixed include directives due to recent directory structure changes

Commit:
Moved some definitions to utility subdirectory

Commit:
Re-added unixtime_from_gps.h file

hangeza committed:
bugfix: moved default argument of constructor of i2cDevice class from implementation section to declaration section where it belongs

hangeza committed:
Merge branch 'v2.0.2' of github.com:MuonPi/muondetector into v2.0.2

hangeza committed:
Fixed severe bug in PigpiodHandler which prevented the pile-up detection and rate throttling to work as intented and resulted in wrong GPIO rates.

hangeza committed:
Added version and build time log output at daemon start

Commit:
Added git hash information to version string

Commit:
Merge branch 'v2.0.2' of github.com:MuonPi/muondetector into v2.0.2

Commit:
Added hash info to package version number

hangeza committed:
Added unique id from eeprom as log parameter. The UID string is now sent as MQTT log message once on daemon startup and log roll-over.

hangeza committed:
Replaced timestring in startup syslog message by the more correct __TIMESTAMP__ macro of the compiler, which gives the real build time, not the time of damon startup as it was before.

hangeza committed:
Added (non-mandatory) "emit" keyword to several lines, where a logParameter signal is triggered. This is cosmetics but makes the intention clearer.

marvin5300 commented:
Does not compile under Windows, linker error with qwt or something about moc files.
[compile_log.txt](https://github.com/MuonPi/muondetector/files/7018307/compile_log.txt)

marvin5300 committed:
Changed find_package for QWT on windows only.

marvin5300 reviewed (dismissed):
Find problem with windows compilation before merging.

- CMake does find QWT: "QWT found: C:/Qwt-6.1.4/lib/qwt.lib"
- include dirs are set: "
    $<BUILD_INTERFACE:${QWT_DIR}/include>
    $<INSTALL_INTERFACE:${QWT_DIR}/include>"

Still I get 'undefined reference' for QwtPlot
I tried to build 'release' and 'debug' with the same result

Commit:
removed build/source directory structure from project.

Commit:
Changed hard coded ip to hostname

marvin5300 committed:
Fixed cmake configuration for windows.
- added 'WIN32' option to add_executable() to hide console window on launch.
- added cmake options for compiling with Qt5.15.2 MinGW_64.
- manually added copying of libstdc++-6.dll, libgcc_s_seh-1.dll and libwinpthread-1.dll during deployment since they won't get copied automatically even if they are needed.

marvin5300 committed:
Merge branch 'v2.0.2' of github.com:MuonPi/muondetector into v2.0.2

Commit:
Removed hash from package version number

Commit:
Merge branch 'v2.0.2' of github.com:MuonPi/muondetector into v2.0.2

marvin5300 committed:
Fixed map by adding libcrypto and libssl to the list of .dll files to deploy.

hangeza commented:
Since all issues have been resolved by now, can we do the merge?
